### PR TITLE
feat: added CDC event stream

### PR DESCRIPTION
**Pull request by mattayres**

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -481,16 +481,17 @@ message DatabaseMetadata {
...
+message StreamRequest {
+  string db = 1;
+  string collection = 2;
+  StreamRequestOptions options = 3;
 }
 
-message StreamChange {
-  string collection_name = 1;
+message StreamEvent {
+  string collection = 1;
```
Comment (himank) [inline server/v1/api.proto:494]: Actually, a stream event can have a bunch of different collections because StreamEvents is on a database level and a transaction can touch different collections. So we need it to be along with the row?

Comment (mattayres) [reply server/v1/api.proto:494]: Each event is a row operation in the transaction with this model. It's worth discussing if that's the right way to go, however it was my understanding of where Yevgeniy wanted to go with the event stream last we spoke. I do like the simplicity of it, though we should include a transaction id so the events can be grouped.

File: server/v1/api.proto [hunk 2]
```diff
@@ -692,7 +693,7 @@ service Tigris {
 
   // Provides real time events for the changes made into the database. Streams by default publish all events for a database
   // but can be restricted to a specific collection. These events can be useful to build event-driven applications.
-  rpc Stream(StreamRequest) returns (stream StreamResponse) {
+  rpc Stream(StreamRequest) returns (stream StreamEvent) {
```
Comment (JigarJoshi) [inline server/v1/api.proto:696]: we are not yet following the proto best practices completely. but it will be great to follow. For example this one here is breaking https://docs.buf.build/lint/rules#rpc_request_standard_name-rpc_response_standard_name-rpc_request_response_unique

tldr; StreamRequest and StreamResponse.

Comment (mattayres) [reply server/v1/api.proto:696]: ok, I'll wrap with a response and keep the event 1:1 with it

Comment (mattayres) [reply server/v1/api.proto]: done now

File: server/v1/api.proto (hunk 1 above)
Comment (himank) [reply server/v1/api.proto:498]: Yes, so we need TxId and also my only concern is in the future may be for some use-case we need to push the events as a single transaction. Moving it inside will allow that.
